src/data/mv_polynomial.lean

```diff
@@ -573,12 +584,12 @@ eval₂_C _ _ _
 @[simp] lemma rename_X (f : β → γ) (b : β) : rename f (X b : mv_polynomial β α) = X (f b) :=
 eval₂_X _ _ _
 
-lemma rename_rename (f : β → γ) (g : γ → δ) (p : mv_polynomial β α) :
+@[simp] lemma rename_rename (f : β → γ) (g : γ → δ) (p : mv_polynomial β α) :
```

Comment:
I don't really like this simp-lemma. If you throw away the `p` argument, you could write the LHS as `(rename g) ∘ (rename f)`. It would then be consistent with the rest of the (category) library to reverse the direction of this statement, and call the lemma `rename_comp`. *Then* it would be a simp-lemma like the rest of the `foo_comp` lemmas.

Reply:
I'm not sure of the correct instincts for simp lemma directions. Here's my reasoning for this one:

This is a lemma about renaming (we've moved on from basic things like composition). Since renaming is the latest and greatest confusing new concept, the notion of "simplification" should be something about reducing the number of appearances of "rename".

The question is whether you think you'll know something about what `rename f` or `rename g` did, or whether you think you'll know something about `g \circ f`.

In any case, I'm happy to just remove the `@[simp]` for now.

Reply:
I'm certainly not an authority when it comes to simp-lemmas. If you don't use the simp-lemma in the rest of the PR, I agree that it might be better to remove the attribute for now.

Reply:
The simp lemma in the other direction won't work because composition is reducible. This direction matches several other simp lemmas for functorial things.

Reply:
Ok, then let's leave it as it is. @semorrison sorry for my complaints.